do/domains.go

```diff
@@ -32,6 +34,21 @@ type DomainRecord struct {
 	*godo.DomainRecord
 }
 
+// A DomainRecordEditRequest is used in place of godo's DomainRecordEditRequest
+// in order to work around the fact that we cannot send a port value of 0 via
+// godo due to Go's JSON encoding logic.
+type DomainRecordEditRequest struct {
+	Type     string `json:"type,omitempty"`
+	Name     string `json:"name,omitempty"`
+	Data     string `json:"data,omitempty"`
+	Priority int    `json:"priority"`
+	Port     *int   `json:"port,omitempty"`
+	TTL      int    `json:"ttl,omitempty"`
+	Weight   int    `json:"weight"`
+	Flags    int    `json:"flags"`
```

Comment:
Let's make it so we hit all of the nullable fields here as pointers so we don't have to come back and change this again if someone decides to make `priority`, `weight`, or `flags` 0.

Reply:
I don't think we should. The only user-facing problem was that SRV records (the only ones that can accept a port) cannot be set to 0, so I wanted this "hack" to have the most narrow scope possible. I think updating all these fields risks creating new issues that we don't want to deal with until we create a godov2.

Reply:
I could go either way on this - if we are going to hack it I feel like we make it the correct hack. I guess we can always change it again when/if we get more issues from users?

Reply:
Yeah, that's what I figured as well – keep the blast radius of the hack small for now, and increase only if we need to. If there're more issues, that's probably an indication that we should prioritize a godo v2 over expanding the hack. Beyond that, there's also some handling of sending a default TTL if the value is 0 that I don't want to mess with 😛